src/harness/reference_models/montecarlo/monteCarloSimITM.py

```diff
@@ -0,0 +1,158 @@
+#    Copyright 2017 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# this function runs the Monte Carlo aggregation Simulation and computes the
```

Comment:
This file makes extensive use fo copy/paste of the internal wf_itm.py code.
Normally you should use it as a blackbox, then you do your aggregation on the result.
Especially since the wf_itm.CalcItmPropagationLoss() can take an a list or an array of reliability, and returns a list of pathloss.

Doing this way avoids duplication and source of errors.

Reply:
The code is built on the reference propagation code. Commscope was assigned to provide the aggregation code. The way you have implemented the ITM code makes it a lot easier to to aggregation.

Reply:
I means, why not simply doing something along the line:

def aggregate(...):
  signal_level = np.zeros(len(devices), N) 
  for k,device in enumerate(devices):
     results = wf_itm.CalcItmPathLoss(.., reliability=np.arange(0.001, 1, res),...)
     path_losses = results.db_loss
     signal_level[k, :] = eirp - np.take(path_losses, np.random.choice(len(path_losses, N) 
  
aggregated_interference = 10*np.log10(np.sum(10**(signal_level/10.), axis=0))

=> And you have your 2000 Monte Carlo samples.

Reply:
Yes, we can modify it as such. Do you think we should keep it in harness/reference_models/propagation/itm/ ? 
Or better still add it as a function to wf_itm.py

Reply:
I dont think it belongs to wf_itm.py whose purpose is to do a set of pathloss calculation for a single device.  
How to aggregate this for multiple devices, taking into account the EIRP,+antenna gain into an RSSI interference level is separate topic.

See my general comment about the need of this module:
 - when doing anything for FSS, PPA, ..., we will simply add the "average" signal level  (ITM or Hybrid) of all devices. Average value is obtained by calling ITM or Hybrid with reliability=-1
 - only for DPA, we need to do MonteCarlo average like you do here. But this is already handled by the DPA reference code under review.

Given all that, I dont think this code will be ever used, except as an example. You could simply add a piece of code in the reference_models/example/ directory, to illustrate how to do MonteCarlo interference aggregation.